### PR TITLE
Add a unit test for BetaBernoulli

### PR DESCRIPTION
**Pull request by ThomasColthurst**
And the makefile rules and documentation for running unit tests in general.

To minimize extra dependencies, I decided to use the Boost unit test library.  As an added bonus, we can now more easily use the other Boost libraries in our non-test code if we want.

### CODE REVIEW

**Review by alex-lew — COMMENTED** [review]
Thanks, this looks great! The boost install + unit test ran smoothly for me. Just made a couple small comments.

File: cxx/distributions/beta_bernoulli.hh [hunk 1]
```diff
@@ -2,7 +2,9 @@
 // See LICENSE.txt
 
 #pragma once
-#include "base.hh"
+#include "globals.hh"
```
[the remark this answers is not in the document]
Comment (ThomasColthurst) [reply cxx/distributions/beta_bernoulli.hh:5]: Fixed.

File: cxx/distributions/base.hh
[the remark this answers is not in the document]
Comment (ThomasColthurst) [reply cxx/distributions/base.hh:1]: I did not.  Removed.

**Review by alex-lew — APPROVED** [review]
Thanks, looks great! Maybe we can also get CI running on Github, so the unit tests at least will be run on every PR.

**Pull request merged**